server/reload_test.go

```diff
@@ -0,0 +1,210 @@
+		t.Fatal("Expected Reload to return an error")
+	}
+
+	// Ensure config didn't change.
+	if !reflect.DeepEqual(golden, server.getOpts()) {
+		t.Fatalf("Options are incorrect.\nexpected: %+v\ngot: %+v",
+			golden, opts)
+	}
+}
+
+// Ensure Reload returns an error when reloading from a bad config file.
+func TestConfigReloadInvalidConfig(t *testing.T) {
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Error getting working directory: %v", err)
+	}
+	config := filepath.Join(dir, "tmp.conf")
+
+	golden := &Options{
+		ConfigFile:     config,
+		Host:           "0.0.0.0",
+		Port:           4222,
+		AuthTimeout:    1.0,
+		Debug:          false,
+		Trace:          true,
+		Logtime:        false,
+		MaxControlLine: 1024,
+		MaxPayload:     1048576,
+		MaxConn:        65536,
+		PingInterval:   2 * time.Minute,
+		MaxPingsOut:    2,
+		WriteDeadline:  2 * time.Second,
+	}
+	processOptions(golden)
+
+	if err := os.Symlink("./configs/reload/test.conf", config); err != nil {
+		t.Fatalf("Error creating symlink: %v", err)
+	}
+	defer os.Remove(config)
+	opts, err := ProcessConfigFile(config)
+	if err != nil {
+		t.Fatalf("Error processing config file: %v", err)
+	}
+	server := New(opts)
+
+	if !reflect.DeepEqual(golden, server.getOpts()) {
+		t.Fatalf("Options are incorrect.\nexpected: %+v\ngot: %+v",
+			golden, opts)
+	}
+
+	// Change config file to bad config by replacing symlink.
+	if err := os.Remove(config); err != nil {
+		t.Fatalf("Error deleting symlink: %v", err)
+	}
+	if err := os.Symlink("./configs/reload/invalid.conf", config); err != nil {
+		t.Fatalf("Error creating symlink: %v", err)
+	}
+
+	// This should fail because the new config should not parse.
+	if err := server.Reload(); err == nil {
+		t.Fatal("Expected Reload to return an error")
+	}
+
+	// Ensure config didn't change.
+	if !reflect.DeepEqual(golden, server.getOpts()) {
+		t.Fatalf("Options are incorrect.\nexpected: %+v\ngot: %+v",
+			golden, opts)
+	}
+}
+
+// Ensure Reload returns nil and the config is changed on success.
+func TestConfigReload(t *testing.T) {
+	// The server package uses a global logger that gets configured by calls to
+	// server.ConfigureLogger(). We need to restore it to prevent side effects.
+	// TODO: Consider getting rid of global logger.
+	logBefore := log
+	defer func() {
+		log = logBefore
+	}()
+
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Error getting working directory: %v", err)
+	}
+	config := filepath.Join(dir, "tmp.conf")
+
+	golden := &Options{
+		ConfigFile:     config,
+		Host:           "0.0.0.0",
+		Port:           4222,
+		AuthTimeout:    1.0,
+		Debug:          false,
+		Trace:          true,
+		Logtime:        false,
+		MaxControlLine: 1024,
+		MaxPayload:     1048576,
+		MaxConn:        65536,
+		PingInterval:   2 * time.Minute,
+		MaxPingsOut:    2,
+		WriteDeadline:  2 * time.Second,
+	}
+	processOptions(golden)
+
+	if err := os.Symlink("./configs/reload/test.conf", config); err != nil {
```

Comment:
@ColinSullivan1 Thanks for checking that. I can add a note to the README if we think this is acceptable. Since this is just for development purposes, this seems OK to me. What do you guys think?